src/common/libutil/base256.c

```diff
@@ -0,0 +1,149 @@
+    "🙂", "🤗", "🤔", "😐", "😑", "😶", "🙄", "😏", "😣", "😥", "😮", "🤐", "😯", "😪", "😫", "😴",
+    "😌", "🤓", "😛", "😜", "😝", "😒", "😓", "😔", "😕", "🙃", "🤑", "😲", "🙁", "😖", "😞", "😟",
+    "😤", "😢", "😭", "😦", "😨", "😩", "😬", "😰", "😱", "😳", "😵", "😡", "😠", "😇", "😷", "🤒",
+    "🤕", "😈", "👿", "👹", "👺", "💀", "👻", "👽", "👾", "🤖", "💩", "😺", "😸", "😹", "😻", "😼",
+    "😽", "🙀", "😿", "😾", "🙈", "🙉", "🙊", "👦", "👧", "👨", "👩", "👴", "👵", "👶", "👼", "👮",
+    "🕵", "💂", "👷", "👳", "👱", "🎅", "👸", "👰", "👲", "🙍", "🙎", "🙅", "🙆", "💁", "🙋", "🙇",
+    "💆", "💇", "🚶", "🏃", "💃", "👯", "🕴", "🗣", "👤", "👥", "🏇", "🏂", "🏌", "🏄", "🚣", "🏊",
+    "🏋", "🚴", "🚵", "🏎", "🏍", "👫", "👬", "👭", "💏", "💑", "👪", "💪", "👈", "👉", "👆", "🖕",
+    "👇", "🖖", "🤘", "🖐", "👌", "👍", "👎", "👊", "👋", "👏", "👐", "🙌", "🙏", "💅", "👂", "👃",
+    "👣", "👀", "👁", "👅", "👄", "💋", "💘", "💓", "💔", "💕", "💖", "💗", "💙", "💚", "💛", "💜",
+    "💝", "💞", "💟", "💌", "💤", "💢", "💣", "💥", "💦", "💨", "💫", "💬", "🗨", "🗯", "💭", "🕳",
+    "👓", "🕶", "👔", "👕", "👖", "👗", "👘", "👙", "👚", "👛", "👜", "👝", "🛍", "🎒", "👞", "👟",
+    "👠", "👡", "👢", "👑", "👒", "🎩", "🎓", "📿", "💄", "💍", "💎", "🐵", "🐒", "🐶", "🐕", "🐩",
+    "🐺", "🐱", "🐈", "🦁", "🐯", "🐅", "🐆", "🐴", "🐎", "🦄", "🐮", "🐂", "🐃", "🐄", "🐷", "🐖",
+    "🐗", "🐽", "🐏", "🐑", "🐐", "🐪", "🐫", "🐘", "🐭", "🐁", "🐀", "🐹", "🐰", "🐇", "🐿", "🐻"
+}, {
+    "🐨", "🐼", "🐾", "🦃", "🐔", "🐓", "🐣", "🐤", "🐥", "🐦", "🐧", "🕊", "🐸", "🐊", "🐢", "🐍",
+    "🐲", "🐉", "🐳", "🐋", "🐬", "🐟", "🐠", "🐡", "🐙", "🐚", "🦀", "🐌", "🐛", "🐜", "🐝", "🐞",
+    "🕷", "🕸", "🦂", "💐", "🌸", "💮", "🏵", "🌹", "🌺", "🌻", "🌼", "🌷", "🌱", "🌲", "🌳", "🌴",
+    "🌵", "🌾", "🌿", "🍀", "🍁", "🍂", "🍃", "🍇", "🍈", "🍉", "🍊", "🍋", "🍌", "🍍", "🍎", "🍏",
+    "🍐", "🍑", "🍒", "🍓", "🍅", "🍆", "🌽", "🌶", "🍄", "🌰", "🍞", "🧀", "🍖", "🍗", "🍔", "🍟",
+    "🍕", "🌭", "🌮", "🌯", "🍳", "🍲", "🍿", "🍱", "🍘", "🍙", "🍚", "🍛", "🍜", "🍝", "🍠", "🍢",
+    "🍣", "🍤", "🍥", "🍡", "🍦", "🍧", "🍨", "🍩", "🍪", "🎂", "🍰", "🍫", "🍬", "🍭", "🍮", "🍯",
+    "🍼", "🍵", "🍶", "🍾", "🍷", "🍸", "🍹", "🍺", "🍻", "🍽", "🍴", "🔪", "🏺", "🌍", "🌎", "🌏",
+    "🌐", "🗺", "🗾", "🏔", "🌋", "🗻", "🏕", "🏖", "🏜", "🏝", "🏞", "🏟", "🏛", "🏗", "🏘", "🏙",
+    "🏚", "🏠", "🏡", "🏢", "🏣", "🏤", "🏥", "🏦", "🏨", "🏩", "🏪", "🏫", "🏬", "🏭", "🏯", "🏰",
+    "💒", "🗼", "🗽", "🕌", "🕍", "🕋", "🌁", "🌃", "🌄", "🌅", "🌆", "🌇", "🌉", "🌌", "🎠", "🎡",
+    "🎢", "💈", "🎪", "🎭", "🖼", "🎨", "🎰", "🚂", "🚃", "🚄", "🚅", "🚆", "🚇", "🚈", "🚉", "🚊",
+    "🚝", "🚞", "🚋", "🚌", "🚍", "🚎", "🚐", "🚑", "🚒", "🚓", "🚔", "🚕", "🚖", "🚗", "🚘", "🚙",
+    "🚚", "🚛", "🚜", "🚲", "🚏", "🛣", "🛤", "🚨", "🚥", "🚦", "🚧", "🚤", "🛳", "🛥", "🚢", "🛩",
+    "🛫", "🛬", "💺", "🚁", "🚟", "🚠", "🚡", "🚀", "🛰", "🛎", "🚪", "🛌", "🛏", "🛋", "🚽", "🚿",
+    "🛀", "🛁", "🕰", "🌞", "🌝", "🌚", "🌑", "🌒", "🌓", "🌔", "🌕", "🌖", "🌗", "🌘", "🌜", "🌛"
+}};
+
+int base256_encode (char *buf,
+                    int buflen,
+                    void *data,
+                    int datalen)
+{
+    register unsigned char table = 0;
+    int i = 0;
+    int n = 0;
+
+    if (buf == NULL || data == NULL || buflen <= 0 || datalen < 0) {
+        errno = EINVAL;
+        return -1;
+    }
```

Comment:
should make make sure datalen can hold the encoded amount via `BASE256_ENCODED_SIZE`?  like what if datalen % 4 != 0?  Oh ... and what if buf = 1 and datalen = 1, then the memcpy right below could overflow.

Reply:
oh i now see that you made a comment in the header, I know this is just a util library for a unique purpose, maybe its overkill